tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,463 @@
+`~zod ran the command` should be displayed in bold green text, signifying that
+the command originated locally.
+
+Now we will connect the sessions. Switch `~zod` back to dojo and enter `|link
+~nus %shoe`. If this succeeds you will see the following.
+```
+>=
+; ~nus is your neighbor
+[linked to [p=~nus q=%shoe]]
+```
+Now `~zod` will have two `%shoe` sessions running - one local one on `~zod` and
+one remote one on `~nus`, which you can access by pressing `Ctrl-X` until you see
+`~nus:shoe>` from `~zod`'s console. On the other hand, you should not see
+`~zod:shoe>` on `~nus`'s side, since you have not connected `~nus` to `~zod`'s
+`%shoe` app. When you enter `demo` from `~nus:shoe>` on
+`~zod`'s console you will again see `~zod ran the command`, but this time it
+should be in the ordinary font used by the console, signifying that the command
+is originating from a remote session. Contrast this with entering `demo` from
+`~nus:shoe>` in `~nus`'s console, which will display `~nus ran the command` in
+bold green text.
+
+Now try to link to `~zod`'s `%shoe` session from `~nus` by switching to the dojo
+on `~nus` and entering `|link ~zod %shoe`. You should see
+```
+>=
+[unlinked from [p=~zod q=%shoe]]
+```
+and if you press `Ctrl-X` you will not get a `~zod:shoe>` prompt. This is
+because the example app is set up to always allow `~zod` to connect (as well as
+subject moons if the ship happens to be a planet) but not `~nus`, so this
+message means that `~nus` failed to connect to `~zod`'s `%shoe` session.
+
+### `%shoe`'s code
+
+```hoon
+::  shoe: example usage of /lib/shoe
+::
+::    the app supports one command: "demo".
+::    running this command renders some text on all sole clients.
+::
+/+  shoe, verb, dbug, default-agent
+```
+
+`/+` is the Ford rune which imports libraries from the `/lib` directory into
+the subject.
+ * `shoe` is the `shoe` library.
+ * `verb` is a library used to print what a Gall agent is doing.
+ * `dbug` is a library of debugging tools. Why do we need this?
+ * `default-agent` contains a Gall agent core with minimal implementations of
+   required Gall arms.
+
+```hoon
+|%
++$  state-0  [%0 ~]
++$  command  ~
+::
++$  card  card:shoe
+--
+```
+The types used by the app.
+
+`$state-0` stores the state of the app, which is null as there is no state to
+keep track of. It is good practice (required?) to have a type for state anyways
+in case the app is made stateful at a later time, and indeed we will do this
+when we modify this app later in the tutorial.
+
+`$command` is typically a set of tagged union types that are possible
+commands that can be entered by the user. Since this app only supports one
+command, it is unnecessary for it to have any associated data, thus the command
+is represented by `~`.
+
+In a non-trivial context, a `$command` is given by `[%name
+data]`, where `%name` is the identifier for the type of command and `data` is
+a type or list of types that contain data needed to execute the command. See
```

Comment:
This is most common, but not strictly true. You could imagine something like `[chat-room=@t =action]` or whatever.